modules/runtime-api/src/commonMain/kotlin/org/jetbrains/compose/storytale/StoryParameterDelegate.kt

```diff
@@ -11,13 +10,17 @@ class StoryParameterDelegate<T>(
     private val type: KClass<*>,
     private val defaultValue: T,
 ) {
-    @Composable
+    @Suppress("UNCHECKED_CAST")
     operator fun getValue(thisRef: Any?, property: KProperty<*>): T = story.nameToParameterMapping.getValue(property.name).state.value as T
 
-    @Composable
+    @Suppress("UNCHECKED_CAST")
+    operator fun setValue(thisRef: Any?, property: KProperty<*>, value: T) {
+        (story.nameToParameterMapping.getValue(property.name).state as MutableState<T>).value = value
+    }
+
     operator fun provideDelegate(thisRef: Any?, property: KProperty<*>) = also {
         story.nameToParameterMapping.getOrPut(property.name) {
-            StoryParameter(property.name, type) { remember { mutableStateOf(defaultValue) } }
+            StoryParameter(property.name, type) { mutableStateOf(defaultValue) }
```

Comment:
I'd like to know (just for my education with Compose). Why don't we need `remember` here?

Reply:
This scope is not Composable anymore. It donesn't have to be composable at all. Perhaps it was made Composable  just to call `@Composable fun remember`.

We use remember only when a state is is created within the Composition. But in this case the state is created once, it's stored in a map `story.nameToParameterMapping`, so no need to "remember" it in the Composition.